### PR TITLE
cargo-outdated: remove `rust` dependency except for building

### PR DESCRIPTION
**Pull request by rami3l**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

cc #124998

This PR applies the fix from #134064 to `cargo-outdated`.

### CODE REVIEW

**Comment by rami3l** [comment]
This is... weird. At least doesn't seem related to my particular fix 🤔️

File: Formula/cargo-outdated.rb [hunk 1]
```diff
@@ -16,9 +16,9 @@ class CargoOutdated < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "30bc7667a6e1bf3e04835e0db2426b25d2bbbc0290140d75ec7200b85f3e002e"
   end
 
+  depends_on "rust" => [:build, :test]
```
Comment (carlocab) [inline Formula/cargo-outdated.rb:19]:
```suggestion
  depends_on "pkg-config" => :build
  depends_on "rust" => [:build, :test]
```

Comment (rami3l) [reply Formula/cargo-outdated.rb:20]: Done.

Comment (rami3l) [reply Formula/cargo-outdated.rb:20]: @carlocab There's still a warning when building on macOS 11 ARM:

```
Full linkage --cached --test --strict cargo-outdated output
  Undeclared dependencies with linkage:
    openssl@3
```

Is there anything to worry about here? Thanks!

Comment (carlocab) [reply Formula/cargo-outdated.rb:20]: Yes, this will break things. Let me have a look...

Comment (carlocab) [reply Formula/cargo-outdated.rb:20]: Try
```diff
diff --git a/Formula/cargo-outdated.rb b/Formula/cargo-outdated.rb
index 0689dd562ae..428a9e54158 100644
--- a/Formula/cargo-outdated.rb
+++ b/Formula/cargo-outdated.rb
@@ -21,6 +21,8 @@ class CargoOutdated < Formula
   depends_on "rust"
 
   def install
+    ENV["OPENSSL_NO_VENDOR"] = "1"
+    ENV["OPENSSL_DIR"] = Formula["openssl@1.1"].opt_prefix
     system "cargo", "install", *std_cargo_args
   end
```

**Review by carlocab — CHANGES_REQUESTED** [review]
This shouldn't link with `openssl@3`.

File: Formula/cargo-outdated.rb (hunk 1 above)
Comment (rami3l) [reply Formula/cargo-outdated.rb]: Done.

**Comment by rami3l** [comment]
@carlocab Thanks a lot for your help!

The fix seems to be working, and I have squashed the commits.
If all goes well with the CI, I'll be making similar changes to the rest of my PRs on your approval.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request merged**